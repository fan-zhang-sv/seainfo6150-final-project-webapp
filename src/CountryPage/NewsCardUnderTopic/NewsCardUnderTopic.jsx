import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import styles from "./NewsCardUnderTopic.module.css";

const NewsCardUnderTopic = (props) => {
  var data = {
    name: props.news["name"],
    provider: props.news["provider"][0]["name"],
    url: props.news["url"],
    imgUrl:
      props.news["image"] && props.news["image"]["contentUrl"]
        ? props.news["image"]["contentUrl"]
        : "",
    description: props.news["description"],
  };
  data = JSON.stringify(data);
  data = data.toString().replace(/%/g, "~~pct~~");
  data = encodeURIComponent(data);
  var path = `/news/${data}`;

  return (
    <Link to={path} className={styles.NewsCard}>
      {props.news["image"] ? (
        <img
          className={styles.Image}
          src={props.news["image"]["thumbnail"]["contentUrl"]}
          alt={props.title}
        ></img>
      ) : null}
      <h4>{props.news["name"]}</h4>
    </Link>
  );
};

NewsCardUnderTopic.propTypes = {
  news: PropTypes.object.isRequired,
};

export default NewsCardUnderTopic;
