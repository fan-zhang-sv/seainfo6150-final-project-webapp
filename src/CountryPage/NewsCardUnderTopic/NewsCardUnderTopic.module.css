.NewsCard {
  /* width: min(85%, 250px); */
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: flex-start;
  text-decoration: none;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  padding: 14px;
  margin: 10px;
}

.Image {
  object-fit: cover;
  width: 5rem;
  height: 5rem;
  margin-right: 10px;
}

.NewsCard:hover {
  box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
}

.NewsCard h4 {
  font-family: "DM Serif Display", serif;
  margin: 6px;
  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  color: black;
}
