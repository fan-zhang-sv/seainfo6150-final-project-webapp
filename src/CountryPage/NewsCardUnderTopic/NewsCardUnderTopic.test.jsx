import React from "react";
import NewsCardUnderTopic from "./NewsCardUnderTopic";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

const news = {
  name: "FDA authorizes 1st COVID-19 vaccine in United States ",
  url:
    "https://www.msn.com/en-us/health/medical/fda-authorizes-1st-covid-19-vaccine-in-united-states/ar-BB1bRB1S",
  image: {
    thumbnail: {
      contentUrl:
        "https://www.bing.com/th?id=OVFT.-pPAb7rjkmMssGiyXjXE0S&pid=News",
      width: 1200,
      height: 630,
    },
    isLicensed: true,
  },
  description:
    "The Food and Drug Administration authorized the United States' first COVID-19 vaccine for people age 16 and over late Friday, a historic moment in a pandemic that has killed over 290,000 in this",
  provider: [
    {
      _type: "Organization",
      name: "ABC News",
      image: {
        thumbnail: {
          contentUrl:
            "https://www.bing.com/th?id=AR_0cfcddaa71cae000b74e3bf52bc84d78",
        },
      },
    },
  ],
  datePublished: "2020-12-12T04:29:00.0000000Z",
};

describe("NewsCardUnderTopic tests", () => {
  it("renders correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <NewsCardUnderTopic news={news} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
