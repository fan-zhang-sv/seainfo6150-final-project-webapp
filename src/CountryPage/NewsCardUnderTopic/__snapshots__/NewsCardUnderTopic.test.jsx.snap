// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsCardUnderTopic tests renders correctly 1`] = `
<div>
  <a
    class="NewsCard"
    href="/news/{\\"name\\"%3A\\"FDA authorizes 1st COVID-19 vaccine in United States \\"%2C\\"provider\\"%3A\\"ABC News\\"%2C\\"url\\"%3A\\"https%3A%2F%2Fwww.msn.com%2Fen-us%2Fhealth%2Fmedical%2Ffda-authorizes-1st-covid-19-vaccine-in-united-states%2Far-BB1bRB1S\\"%2C\\"imgUrl\\"%3A\\"\\"%2C\\"description\\"%3A\\"The Food and Drug Administration authorized the United States' first COVID-19 vaccine for people age 16 and over late Friday%2C a historic moment in a pandemic that has killed over 290%2C000 in this\\"}"
  >
    <img
      class="Image"
      src="https://www.bing.com/th?id=OVFT.-pPAb7rjkmMssGiyXjXE0S&pid=News"
    />
    <h4>
      FDA authorizes 1st COVID-19 vaccine in United States 
    </h4>
  </a>
</div>
`;
